discord-bot/src/interactions/application_commands/matchmaking.rs

```diff
@@ -237,6 +264,153 @@ impl InteractionHandler for MatchmakingCommandHandler {
+                let opponent = resolved
+                    .users
+                    .values()
+                    .next()
+                    .ok_or_else(|| anyhow!("cannot get the user specified in \"report-score\" command"))?;
+                
+                let response = self.utils
+                    .http_client
+                    .channel(chan_id)
+                    .exec()
+                    .await?;
+
+                if !response.status().is_success() {
+                    return Err(anyhow!("failed to get thread"));
+                }
+
+                let channel:Channel = response
+                    .model()
+                    .await?;
+
+                let thread_members:Vec<ThreadMember> = self.utils
+                    .http_client
+                    .thread_members(channel.id)
+                    .exec()
+                    .await?
+                    .models()
+                    .await?;
+
+                let mut reporter_is_member:bool = false;
+                let mut opponent_is_member:bool = false;
+                
+                for member in thread_members {
+                    let member_user_id = member.user_id.ok_or_else(|| anyhow!("cannot get the user specified in \"report-score\" command"))?;
+                    if user.id == member_user_id {
+                        reporter_is_member = true;
+                    } else if opponent.id == member_user_id {
+                        opponent_is_member = true;
+                    } else if reporter_is_member && opponent_is_member {
+                        break;
+                    }
+                }
+
+                // TODO: Check if opponent is the bot, because that would be invalid.
+                if !reporter_is_member {
+                    return Err(anyhow!("user isn't part of the lobby"));
+                } else if !opponent_is_member {
+                    return Err(anyhow!("opponent isn't part of the lobby"));
+                }
+
+                let options:Vec<CommandDataOption>;
+                match subcommand.value.clone() {
+                    CommandOptionValue::SubCommand(x) => options = x,
+                    _ => options = Vec::<CommandDataOption>::new()
+                }
+
+                let wins_option_value = options
+                    .get(1)
+                    .ok_or_else(|| anyhow!("could not get wins option"))?
+                    .value
+                    .clone();
+                
+                let loses_option_value = options
+                    .get(2)
+                    .ok_or_else(|| anyhow!("could not get loses option"))?
+                    .value
+                    .clone();
+                
+                let wins;
+                match wins_option_value {
+                    CommandOptionValue::Integer(x) => wins = x as i64,
+                    _ => wins = 0
+                }
```

Comment:
As far as code style is concerned, I would prefer the following:
```suggestion
                let wins = match wins_option_value {
                    CommandOptionValue::Integer(x) => x as i64,
                    _ => 0
                }
```

Reply:
Oh I didn't thought of that being possible, I'll change both of those then.